src/main/clojure/conexp/fca/factorization.clj

```diff
@@ -0,0 +1,927 @@
+    (:require [conexp.base :refer :all]
+              [conexp.fca.fast :refer [to-binary-matrix]]
+              [conexp.fca.contexts :refer :all]))
+
+(defprotocol factorization-protocol
+  (context [this] "Returns the context that is generated after using the algorithm algo k-times on it")
+  (object-factor [this] "Returns the object-factor of the context generated by the algorithm")
+  (attribute-factor [this] "Returns the attribute-factor of the context generated by the algorithm"))
+
+(defrecord factorization-record [context object-factor attribute-factor]
+  factorization-protocol
+  (context [this] context)
+  (object-factor [this] object-factor)
+  (attribute-factor [this] attribute-factor))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; helper functions
+
+(defn make-matrix-from-concept
+  [a b n m]
+  ;; make matrix from concept a&b with n m 
+  (loop [i 0 j 0 matrix []]
+    (if (> (inc j) n)
+      matrix
+      (recur
+       (cond (>= (inc i) m) 0 :else (inc i))
+       (cond (>= (inc i) m) (inc j) :else j)
+       (cond (and
+              (some #(= j %) a)
+              (some #(= i %) b))
+             (conj matrix 1) :else (conj matrix 0))))))
+
+(defn transpose
+  "Transposes matrix"
+  [a]
+  (into [] (apply map vector a)))
+
+(defn make-matrix-from-context
+  "Converts context into binary Matrix"
+  [context]
+  (let [stringContext (clojure.string/split-lines context)]
+    (loop [i 2
+           curLine (nth (clojure.string/split (nth stringContext i) #"([|])") 1)
+           matrix (conj [] ((fn [line] (into [] (for [s line] (if (= s "x") 1 0)))) (re-seq #"\S+" curLine)))]
+      (if (<= (count stringContext) i)
+        (subvec (conj matrix ((fn [line] (into [] (for [s line] (if (= s "x") 1 0)))) (re-seq #"\S+" curLine))) 2)
+        (recur
+         (inc i)
+         (nth (clojure.string/split (nth stringContext i) #"([|])") 1)
+         (conj matrix ((fn [line] (into [] (for [s line] (if (= s "x") 1 0)))) (re-seq #"\S+" curLine))))))))
+
+(defn normalize-context
+  [context]
+  (make-context-from-matrix
+   (first (context-size context))
+   (second (context-size context))
+   (into [] (flatten (make-matrix-from-context (context-to-string context))))))
+
+;; Creates for the grecond algorithm usable data from the conexp.fca.contexts/concepts method
+(defn- grecond-create-usable
+  [S]
+  (loop [i 0 c []]
+    (if (> i (- (count S) 1))
+      c
+      (recur
+       (inc i)
+       (cond
+         (not= 0 (* (count (nth (nth S i) 0)) (count (nth (nth S i) 1))))
+         (conj c (nth S i))
+         :else c)))))
+
+;; Unite-Operator for 2 matrices
+(defn- unite
+  [A B]
+  (loop [i 0 C []]
+    (if (>= i (count A))
+      C
+      (recur
+       (inc i)
+       (conj C (into [] (map (fn [x y] (if (or (= 1 x) (= 1 y)) 1 0)) (nth A i) (nth B i))))))))
+
+(defn convert-context-to-matrix
+  [context]
+  (into [] (flatten (make-matrix-from-context (context-to-string (normalize-context context))))))
```

Comment:
The dependence on context-to-string to create an incidence matrix may case side effects. Maybe a representation as used in the fast namespace is better.